inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -515,6 +541,29 @@ public void writeMessage(InputStream message) {
           statsTraceCtx.outboundMessageSent(outboundSeqNo, -1, -1);
           clientStream.statsTraceCtx.inboundMessage(outboundSeqNo);
           clientStream.statsTraceCtx.inboundMessageRead(outboundSeqNo, -1, -1);
+
+          long messageLength;
+          try {
+            if (assumedMessageSize != -1) {
+              messageLength = assumedMessageSize;
+            } else if (message instanceof KnownLength || message instanceof ByteArrayInputStream) {
+              messageLength = getKnownLength(message);
+            } else {
+              InputStream oldMessage = message;
+              byte[] payload = ByteStreams.toByteArray(message);
+              messageLength = payload.length;
+              message = new ByteArrayInputStream(payload);
+              oldMessage.close();
+            }
+          } catch (IOException e) {
+            throw new RuntimeException("Error processing the message length", e);
```

Comment:
I'm uncertain about what "proper" exception handling looks like here. I'll need to look into it.

Reply:
Looks like any exception is fine. It might be converted into CANCELLED:
https://github.com/grpc/grpc-java/blob/d034a56cb0de3b5e3635c8ecb381085d02b4ddbf/core/src/main/java/io/grpc/internal/ClientCallImpl.java#L526

Although that isn't guaranteed; it looks like when using DelayedStream nothing might catch this exception. I think the answer here is "the exception never happens so nobody would notice if it isn't handled well."